seafile/SeafActivityViewController.m

```diff
@@ -421,13 +475,27 @@ - (void)getCommitModificationDetail:(NSString *)repoId url:(NSString *)url fromC
 }
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
+    [tableView deselectRowAtIndexPath:indexPath animated:true];
     if (indexPath.row >= _events.count)
         return;
 
     NSDictionary *event = [_events objectAtIndex:indexPath.row];
     NSString *etype = [event objectForKey:@"etype"];
-    if (![etype isEqualToString:@"repo-update"])
-        return;
+    if ([_connection isNewActivitiesApiSupported]) {
+        NSString *name = [event valueForKey:@"name"];
+        NSString *op_type = [event valueForKey:@"op_type"];
+        NSString *obj_type = [event valueForKey:@"obj_type"];
+        if ([obj_type containsString:@"draft"] && [op_type isEqualToString:@"publish"]) {
+            return [self openFile:[event valueForKey:@"path"] inRepo:[event valueForKey:@"repo_id"]];
+        } else if ([op_type isEqualToString:@"delete"] || [op_type isEqualToString:@"clean-up-trash"] || [name containsString:@"(draft).md"]) {
+            return;
+        }
+    } else {
```

Comment:
if it always return something in the if clause, else is not necessary here, you can remove else to avoid too many nested levels.